home/utils/availability.py

```diff
@@ -0,0 +1,387 @@
+    Args:
+        user1: First CustomUser instance
+        user2: Second CustomUser instance
+
+    Returns:
+        Tuple of (overlapping_slots, hour_blocks_count)
+        - overlapping_slots: Sorted list of time slot values where both users overlap
+        - hour_blocks_count: Number of complete 1-hour blocks
+    """
+    user1_slots = set(get_user_slots(user1))
+    user2_slots = set(get_user_slots(user2))
+
+    overlapping_slots = user1_slots.intersection(user2_slots)
+    sorted_overlap = sorted(overlapping_slots)
+
+    hour_blocks = count_one_hour_blocks(sorted_overlap)
+
+    return sorted_overlap, hour_blocks
+
+
+def calculate_team_overlap(
+    navigator_users: list[Any],
+    captain_user: Any | None,
+    djangonaut_users: list[Any],
+) -> dict[str, Any]:
+    """
+    Calculate availability overlaps for an entire team.
+
+    Calculates:
+    1. Navigator meeting hours: All navigators + all djangonauts together
+    2. Captain 1-on-1 hours: Captain with each individual djangonaut
+
+    Args:
+        navigator_users: List of navigator CustomUser instances
+        captain_user: Captain CustomUser instance (can be None)
+        djangonaut_users: List of djangonaut CustomUser instances
+
+    Returns:
+        Dictionary with:
+        - navigator_meeting_slots: List of overlapping time slots for navigator meetings
+        - navigator_meeting_hours: Number of 1-hour blocks for navigator meetings
+        - captain_meetings: List of dicts with djangonaut info and overlap data
+        - is_valid: Boolean indicating if team meets minimum requirements (5 hours)
+    """
+    result = {
+        "navigator_meeting_slots": [],
+        "navigator_meeting_hours": 0,
+        "captain_meetings": [],
+        "is_valid": False,
+    }
+
+    # Calculate navigator meeting overlap (navigators + djangonauts, no captain)
+    navigator_meeting_participants = navigator_users + djangonaut_users
+    if navigator_meeting_participants:
+        nav_slots, nav_hours = calculate_group_overlap(navigator_meeting_participants)
+        result["navigator_meeting_slots"] = nav_slots
+        result["navigator_meeting_hours"] = nav_hours
+        result["is_valid"] = nav_hours >= 5
+
+    # Calculate captain 1-on-1 overlaps with each djangonaut
+    if captain_user and djangonaut_users:
+        captain_meetings = []
+        for djangonaut in djangonaut_users:
+            slots, hours = calculate_pairwise_overlap(captain_user, djangonaut)
+            captain_meetings.append(
+                {
+                    "djangonaut": djangonaut,
+                    "slots": slots,
+                    "hours": hours,
+                }
+            )
+        result["captain_meetings"] = captain_meetings
+
+        # Optional: Add warning if any captain meeting has < 2 hours
+        min_captain_hours = min(
+            (meeting["hours"] for meeting in captain_meetings), default=0
+        )
+        result["min_captain_hours"] = min_captain_hours
```

Comment:
Remove this comment. Instead add logic to mark the team as invalid if there's fewer than 3 hours of overlap for any Djangonaut. Make this a constant too in the same place as the 5 hours for navigator.